90metalmdsquash/metal-md-disks.sh

```diff
@@ -9,17 +9,19 @@ ls /dev/sd* > /dev/null 2>&1 || exit 1
 type metal_die > /dev/null 2>&1 || . /lib/metal-lib.sh
 type pave > /dev/null 2>&1 || . /lib/metal-md-lib.sh
 
-# PRELIMINARY SCAN
+# PRELIMINARY SCAN; this jumpstarts any existing RAIDs on regular "non-deployment boots." 
+# EXIT 0 if no metal-server is set; exit nicely after the preliminary RAID scan, this is a "non-deployment boot"
 /sbin/metal-md-scan
 [ -z "${metal_server:-}" ] && exit 0
 
-# DISKS or RETRY
-md_disks="$(lsblk -l -o SIZE,NAME,TYPE,TRAN | grep -E '(sata|nvme|sas)' | sort -h | awk '{print $2}' | head -n ${metal_disks} | tr '\n' ' ')"
-[ -z "${md_disks}" ] && exit 1
-
 # PAVE & GO-AROUND/RETRY
 [ ! -f /tmp/metalpave.done ] && [ "${metal_nowipe:-0}" != 1 ] && pave
 
+# DISKS; disks were detected, find the amount we need or die. Also die if there are 0 disks.
+# MAINTAINER NOTE: We filter out NVME partitions because they'll exist at this point since pave() is called afterwards.
+md_disks="$(lsblk -l -o SIZE,NAME,TYPE,TRAN | grep -E '('"$metal_transports"')' | sort -u | awk '{print $2}' | head -n ${metal_disks} | tr '\n' ' ' | sed 's/ *$//')"
```

Comment:
The change of `sort -u` to `sort -h` is breaking the sorting of the disks by _size_ and the RAID is going to grab the wrong disks.

```bash
+ lsblk
NAME                MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
loop0                 7:0    0   4.7G  1 loop  /run/rootfsbase
loop1                 7:1    0   3.2G  0 loop
`-live-overlay-pool 254:0    0    32G  0 dm
loop2                 7:2    0    32G  0 loop
`-live-overlay-pool 254:0    0    32G  0 dm
sda                   8:0    0 447.1G  0 disk
|-sda1                8:1    0   476M  0 part
| `-md127             9:127  0 475.9M  0 raid1
|-sda2                8:2    0  22.8G  0 part
| `-md126             9:126  0  22.8G  0 raid1 /run/initramfs/live
|-sda3                8:3    0 139.7G  0 part
| `-md125             9:125  0 139.6G  0 raid1 /run/initramfs/overlayfs
`-sda4                8:4    0 139.7G  0 part
  `-md124             9:124  0 139.6G  0 raid1
sdb                   8:16   0   1.7T  0 disk
|-sdb1                8:17   0   476M  0 part
| `-md127             9:127  0 475.9M  0 raid1
|-sdb2                8:18   0  22.8G  0 part
| `-md126             9:126  0  22.8G  0 raid1 /run/initramfs/live
|-sdb3                8:19   0 139.7G  0 part
| `-md125             9:125  0 139.6G  0 raid1 /run/initramfs/overlayfs
`-sdb4                8:20   0 139.7G  0 part
  `-md124             9:124  0 139.6G  0 raid1
sdc                   8:32   0 447.1G  0 disk
```
Observe above that sdb is the larger, ephemeral disk. However it has been adopted as a RAID member.